FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,51 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+
+			guard self != nil else { return }
+
+			switch result {
+			case let .success((data, response)):
+
+				completion(FeedItemMapper.map(data, response: response))
+
+			case .failure:
+				completion(.failure(Error.connectivity))
+			}
+		}
+	}
+}
+
+internal final class FeedItemMapper {
```

Comment:
`FeedItemMapper` is an implementation detail of `RemoteFeedLoader`, and since it's in the same file in should be marked as `private`:
```suggestion
private final class FeedItemMapper {
```

Reply:
`FeedItemMapper` is defined as a class, so we can instantiate it as `FeedItemMapper()`. However, all its methods are static - it has no instance methods or properties.

It may be confusing for other developers to be able to instantiate a class but then not be able to do anything with it.

As you'll see in the following lectures, you can make the initializer private in such cases since the `FeedItemMapper` is just a namespace for static methods, with no instance-specific behavior.

Or you can make it an enum with no cases, which can never be instantiated!

Both solutions will disallow instantiation at compile time so that `FeedItemMapper()` won't compile ✅